analyzers/src/SonarAnalyzer.CSharp/Rules/PropertyToAutoProperty.cs

```diff
@@ -35,158 +35,123 @@ namespace SonarAnalyzer.Rules.CSharp
+                        || HasDifferentModifiers(propertyDeclaration.AccessorList.Accessors)
+                        || HasAttributes(propertyDeclaration.AccessorList.Accessors))
                     {
                         return;
                     }
 
                     var accessors = propertyDeclaration.AccessorList.Accessors;
                     var getter = accessors.FirstOrDefault(a => a.IsKind(SyntaxKind.GetAccessorDeclaration));
-                    var setter = accessors.FirstOrDefault(a => a.IsKind(SyntaxKind.SetAccessorDeclaration));
-
-                    if (getter == null || setter == null)
-                    {
-                        return;
-                    }
-                    if (TryGetFieldFromGetter(getter, c.SemanticModel, out var getterField) &&
-                        TryGetFieldFromSetter(setter, c.SemanticModel, out var setterField) &&
-                        getterField.Equals(setterField) &&
-                        !getterField.GetAttributes().Any() &&
-                        !getterField.IsVolatile &&
-                        getterField.IsStatic == propertySymbol.IsStatic &&
-                        getterField.Type.Equals(propertySymbol.Type))
+                    var setter = accessors.FirstOrDefault(a => a.IsAnyKind(SyntaxKind.SetAccessorDeclaration, SyntaxKindEx.InitAccessorDeclaration));
+
+                    if (getter != null
+                        && setter != null
+                        && FieldFromGetter(getter, c.SemanticModel) is { } getterField
+                        && FieldFromSetter(setter, c.SemanticModel) is { } setterField
+                        && getterField.Equals(setterField)
+                        && !getterField.GetAttributes().Any()
+                        && !getterField.IsVolatile
+                        && getterField.IsStatic == propertySymbol.IsStatic
+                        && getterField.Type.Equals(propertySymbol.Type))
                     {
-                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, propertyDeclaration.Identifier.GetLocation()));
+                        c.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, propertyDeclaration.Identifier.GetLocation()));
                     }
                 },
                 SyntaxKind.PropertyDeclaration);
-        }
 
-        private static bool HasAttributes(SyntaxList<AccessorDeclarationSyntax> accessors)
-        {
-            return accessors.Any(a => a.AttributeLists.Any());
-        }
+        private static bool HasAttributes(SyntaxList<AccessorDeclarationSyntax> accessors) =>
+            accessors.Any(a => a.AttributeLists.Any());
 
         private static bool HasDifferentModifiers(SyntaxList<AccessorDeclarationSyntax> accessors)
         {
             var accessor1 = accessors.First();
-            var modifiers = GetModifierKinds(accessor1).ToHashSet();
+            var modifiers = ModifierKinds(accessor1).ToHashSet();
```

Comment:
I don't see a reason to give it a name. We don't need the empty line after it as well.
```suggestion
            var modifiers = ModifierKinds(accessors.First()).ToHashSet();
```